src/modules/dashboard/components/Whitelist/CSVUploader/CSVUploader.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react';
+import Papa, { ParseResult } from 'papaparse';
+import { useField } from 'formik';
+import { MessageDescriptor } from 'react-intl';
+import isNil from 'lodash/isNil';
+
+import FileUpload, { UploadFile } from '~core/FileUpload';
+import { isAddress } from '~utils/web3';
+
+import CSVUploaderItem from './CSVUploaderItem';
+
+interface Props {
+  name: string;
+  error?: string | MessageDescriptor;
+  processingData: boolean;
+  setProcessingData: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const MIME_TYPES = ['text/csv'];
+
+const CSVUploader = ({
+  name,
+  error,
+  processingData,
+  setProcessingData,
+}: Props) => {
+  const [CSVFile, setCSVFile] = useState<File | null>(null);
+  const [parsedCSV, setParsedCSV] = useState<ParseResult<unknown> | null>(null);
+  const [, { value }, { setValue }] = useField<UploadFile[]>(name);
+
+  const handleUploadError = async () => {
+    if (isNil(value[0].parsedData)) {
+      await setValue([{ ...value[0], parsedData: [] }]);
+    }
+  };
+
+  useEffect(() => {
+    if (CSVFile && !parsedCSV) {
+      Papa.parse(CSVFile, {
+        complete: setParsedCSV,
+        header: true,
+      });
+    } else if (!CSVFile && parsedCSV) {
+      setParsedCSV(null);
+    }
+  }, [CSVFile, parsedCSV]);
+
+  useEffect(() => {
+    if (parsedCSV && value[0] && isNil(value[0].parsedData)) {
+      const headerExpectedName = 'Whitelist Address';
+      if (
+        parsedCSV.meta.fields?.length === 1 &&
+        parsedCSV.meta.fields[0] === headerExpectedName
+      ) {
+        const validAddresses = parsedCSV.data.flatMap(
+          (CSVRow: Record<string, any>) => {
+            const potentialAddress: string = CSVRow[headerExpectedName];
+            if (potentialAddress) {
+              return isAddress(potentialAddress) ? [potentialAddress] : [];
```

Comment:
We need a dedicated message when one of the addresses in the file is invalid. Something like:
- ```.csv file contains an invalid address`` ... or something along those lines _(maybe ask product)_

I think you can use `setError` available from the `useField` hook to just set that error message manually.